src/Elasticsearch/Document/Document.php

```diff
@@ -12,14 +14,25 @@ class Document implements DocumentInterface
     private $contentType;
     /** @var array<mixed> */
     private $source;
+    /** @var string */
+    private $index;
+    /** @var array<string, mixed> */
+    private $raw;
+    /** @var string|null*/
+    private $highlight;
 
     /**
-     * @param array{_id: string, _type: ?string, _source: array} $document
+     * @param Result|array<string, mixed> $document
      */
-    public function __construct(array $document)
+    public function __construct($document)
```

Comment:
private construct? And to static constructs? ::fromArray, ::fromResult ?

Reply:
It was also an idea, I was not sure what was the better approche, but It like your solution

Reply:
Done

Reply:
Perfect!